sklearn/impute/tests/test_impute.py

```diff
@@ -1786,3 +1786,57 @@ def test_simple_imputer_constant_fill_value_casting():
         )
         X_trans = imputer.fit_transform(X_float32)
         assert X_trans.dtype == X_float32.dtype
+
+
+@pytest.mark.parametrize("strategy", ["mean", "median", "most_frequent", "constant"])
+def test_iterative_imputer_no_empty_features(strategy):
```

Comment:
There is a test called `test_iterative_imputer_keep_empty_features`: we can remove it and make sure that the case that we checked there is also covered by one of the two cases here.

Reply:
Addressed in https://github.com/scikit-learn/scikit-learn/pull/29779/commits/0e1f067d1f7a51f8112e8ebb0d04d0f05916ad88